x-pack/plugins/lens/public/datasources/form_based/datapanel.test.tsx

```diff
@@ -602,441 +425,258 @@ describe.skip('FormBased Data Panel', () => {
+                  meta: true,
+                  type: 'string',
+                  searchable: true,
+                  aggregatable: true,
+                },
+              ],
             },
-          })}
-        />
-      );
-
-      wrapper
-        .find('[data-test-subj="lnsIndexPatternMetaFields"]')
-        .find('button')
-        .first()
-        .simulate('click');
-      expect(
-        wrapper
-          .find('[data-test-subj="lnsIndexPatternMetaFields"]')
-          .find(FieldItem)
-          .first()
-          .prop('field').name
-      ).toEqual('_id');
+          },
+        }),
+      });
+      await userEvent.click(screen.getByText('Meta fields'));
+      expect(getMetaFieldsNames()).toEqual(['_id']);
     });
 
     it('should display NoFieldsCallout when all fields are empty', async () => {
-      (ExistingFieldsServiceApi.loadFieldExisting as jest.Mock).mockImplementation(async () => {
-        return {
-          existingFieldNames: [],
-        };
+      loadFieldExistingMock.mockResolvedValueOnce({
+        existingFieldNames: [],
+        indexPatternTitle: 'idx1',
       });
+      await renderFormBasedDataPanel();
+      expect(getAvailableFieldsNames()).toEqual([]);
+      expect(
+        screen.getByTestId('lnsIndexPatternAvailableFieldsNoFieldsCallout-noFieldsMatch')
+      ).toBeInTheDocument();
 
-      const wrapper = await mountAndWaitForLazyModules(<InnerFormBasedDataPanel {...props} />);
+      await userEvent.click(screen.getByText('Empty fields'));
 
-      expect(wrapper.find(EuiCallOut).length).toEqual(2);
-      expect(
-        wrapper
-          .find('[data-test-subj="lnsIndexPatternAvailableFields"]')
-          .find(FieldItem)
-          .map((fieldItem) => fieldItem.prop('field').name)
-      ).toEqual([]);
-      wrapper
-        .find('[data-test-subj="lnsIndexPatternEmptyFields"]')
-        .find('button')
-        .first()
-        .simulate('click');
-      expect(
-        wrapper
-          .find('[data-test-subj="lnsIndexPatternEmptyFields"]')
-          .find(FieldItem)
-          .map((fieldItem) => (fieldItem.prop('field') as IndexPatternField).displayName)
-      ).toEqual(['amemory', 'bytes', 'client', 'source', 'timestampLabel']);
+      expect(getEmptyFieldsNames()).toEqual([
+        'amemory',
+        'bytes',
+        'client',
+        'source',
+        'timestampLabel',
+      ]);
     });
 
     it('should display spinner for available fields accordion if existing fields are not loaded yet', async () => {
-      let resolveFunction: (arg: unknown) => void;
-      (ExistingFieldsServiceApi.loadFieldExisting as jest.Mock).mockReset();
-      (ExistingFieldsServiceApi.loadFieldExisting as jest.Mock).mockImplementation(() => {
-        return new Promise((resolve) => {
+      let resolveFunction: (arg: {
+        existingFieldNames: string[];
+        indexPatternTitle: string;
+      }) => void;
+      loadFieldExistingMock.mockResolvedValueOnce(
```

Comment:
nit: it is already returning a `Promise`, so no need to wrap it again within a new `Promise` that resolves.

```suggestion
      loadFieldExistingMock. mockReturnValueOnce(
```